include/irr/builtin/glsl/bxdf/common_samples.glsl

```diff
@@ -3,24 +3,37 @@
 
 #include <irr/builtin/glsl/bxdf/brdf/cos_weighted_sample.glsl>
 
-irr_glsl_BSDFSample irr_glsl_transmission_cos_sample(in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
+irr_glsl_BSDFSample irr_glsl_transmission_cos_generate(in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
 {
     irr_glsl_BSDFSample smpl;
     smpl.L = -interaction.isotropic.V.dir;
-    smpl.probability = 1.0;
+    //smpl.probability = 1.0;
     
     return smpl;
 }
 
-irr_glsl_BSDFSample irr_glsl_delta_cos_sample(in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
+irr_glsl_transmission_cos_remainder_and_pdf(out float pdf, in irr_glsl_BSDFAnisotropicParams params, in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
+{
+	pdf = 1.0;
+	return 0.0;
+}
+
+irr_glsl_BSDFSample irr_glsl_delta_cos_generate(in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
 {
     irr_glsl_BSDFSample smpl;
     smpl.L = interaction.isotropic.N*2.0*interaction.isotropic.NdotV - interaction.isotropic.V.dir;
-    smpl.probability = 1.0;
+    //smpl.probability = 1.0;
 
     return smpl;
 }
 
+irr_glsl_delta_cos_remainder_and_pdf(out float pdf, in irr_glsl_BSDFAnisotropicParams params, in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
+{
+	pdf = 1.0;
+	return 0.0;
+}
+
+//this is probably wrong so not touching it
```

Comment:
I'll fix it

Reply:
I need to fix it for 3 different cases:
1) Thidielectric (no wavelength dependent stuff)
2) Approximate Dielectric (compute a single IoR from the weighted IoRs)
3) Pseudo-Spectral Dielectric